DROD/FaceWidget.cpp

```diff
@@ -154,157 +152,199 @@ CFaceWidget::~CFaceWidget()
 }
 
 //****************************************************************************
-SDL_Rect* CFaceWidget::getEyeMask() const {
-	return GetCharacter() == Speaker_Gunthro ? m_GunthroEyeMaskRectArray + this->eMood : m_EyeMaskRectArray + this->eMood;
+SDL_Rect* CFaceWidget::getEyeMask() {
+	const Face* face = GetActiveFace();
+	return face->eCharacter == Speaker_Gunthro ? m_GunthroEyeMaskRectArray + face->eMood : m_EyeMaskRectArray + face->eMood;
 }
 
-const CFaceWidget::POINT* CFaceWidget::getEyeMaskOffset() const {
-	return GetCharacter() == Speaker_Gunthro ? m_GunthroEyeMaskOffsetArray + this->eMood : m_EyeMaskOffsetArray + this->eMood;
+const CFaceWidget::POINT* CFaceWidget::getEyeMaskOffset() {
+	const Face* face = GetActiveFace();
+	return face->eCharacter == Speaker_Gunthro ? m_GunthroEyeMaskOffsetArray + face->eMood : m_EyeMaskOffsetArray + face->eMood;
 }
 
-UINT CFaceWidget::getBetweenPupils() const {
-	if (GetCharacter() != Speaker_Gunthro)
+UINT CFaceWidget::getBetweenPupils() {
+	const Face* face = GetActiveFace();
+	if (face->eCharacter != Speaker_Gunthro)
 		return CX_BETWEEN_PUPILS;
 
-	switch (this->eMood) {
+	switch (face->eMood) {
 		case Mood_Aggressive:
 		case Mood_Strike:
 			return CX_BETWEEN_PUPILS_GUNTHRO_AGG;
 		default: return CX_BETWEEN_PUPILS_GUNTHRO;
 	}
 }
 
+Face* CFaceWidget::GetFace(const FaceWidgetLayer layer) {
+	switch (layer) {
+		case Speaker: return &faceSpeaker;
+		case Death: return &faceDying;
+		case PlayerRole:
+		default:
+			return &facePlayer;
+	}
+}
+
+const FaceWidgetLayer CFaceWidget::GetActiveLayer() const{
+	if (faceSpeaker.bIsActive)
+		return faceSpeaker.eLayer;
+
+	else if (faceDying.bIsActive)
```

Comment:
Can remove the two elses here to make the code a bit cleaner and easier to read.